src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContextFactory.cs

```diff
@@ -14,15 +15,22 @@ namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
     /// </summary>
     internal interface IWorkspaceProjectContextFactory
     {
+        /// <inheritdoc cref="CreateProjectContextAsync"/>
+        [Obsolete("Use CreateProjectContextAsync instead")]
+        IWorkspaceProjectContext CreateProjectContext(string languageName, string projectUniqueName, string projectFilePath, Guid projectGuid, object hierarchy, string binOutputPath);
+
         /// <summary>
-        /// Creates and initializes a new Workspace project and returns a <see cref="IWorkspaceProjectContext"/> to lazily initialize the properties and items for the project.
+        /// Creates and initializes a new Workspace project and returns a <see
+        /// cref="IWorkspaceProjectContext"/> to lazily initialize the properties and items for the
+        /// project.  This method guarantees that either the project is added (and the returned task
+        /// completes) or cancellation is observed and no project is added.
         /// </summary>
         /// <param name="languageName">Project language.</param>
         /// <param name="projectUniqueName">Unique name for the project.</param>
         /// <param name="projectFilePath">Full path to the project file for the project.</param>
         /// <param name="projectGuid">Project guid.</param>
         /// <param name="hierarchy">Obsolete. The argument is ignored.</param>
```

Comment:
❔ Why keep this if it's just going to be ignored?

Reply:
i can't tell i the comment is a lie or not.  the value does seem to be used.  i'd rather not change something that could subtly break things.